HTTPServer.test/MockDirectoryContents.cs

```diff
@@ -0,0 +1,23 @@
+﻿using HTTPServer.core;
+
+namespace HTTPServer.test
+{
+    public class MockDirectoryContents: IDirectoryContents
+    {
+        public string DirectoryPath { get; }
+
+        public MockDirectoryContents(string directoryPath)
+        {
+            DirectoryPath = directoryPath;
+        }
+        public string[] GetFiles()
+        {
+            return new []{ "C:\\gitwork\\HTTP Server\\.gitattributes" , "C:\\gitwork\\HTTP Server\\.gitignore" , "C:\\gitwork\\HTTP Server\\HTTP Server.core.dll" , "C:\\gitwork\\HTTP Server\\HTTPServer.sln" , "C:\\gitwork\\HTTP Server\\server.config" , "C:\\gitwork\\HTTP Server\\server.exe" };
```

Comment:
this is a lot of test data to check and work with. tdd generally has such a quick turnaround cycle because the goal is specify the minimum possible code that will pass the test, including the test data, so that the different cases can be clearly tested. another useful thing that i do is not make my test data look like my prod data to avoid encoding too many assumptions into my implementation. for example, do these strings really need to be full paths?